src/status_im/ui/screens/signing/views.cljs

```diff
@@ -90,14 +92,81 @@
     [react/view {:padding 6}
      [react/text {:style {:color colors/blue}} (i18n/label :t/cancel)]]]])
 
-(views/defview password-view [{:keys [type error in-progress? enabled?]}]
+(views/defview keycard-pin-view []
+  (views/letsubs [pin [:hardwallet/pin]]
+    [react/view
+     [pin.views/pin-view
+      {:pin           pin
+       :retry-counter nil
+       :step          :sign
+       :status        nil
+       :error-label   nil}]]))
+
+(defn- keycard-connect-view []
+  [react/view {:padding-vertical 20
+               :flex             1
+               :align-items      :center
+               :justify-content  :center}
+   [react/image {:source      (resources/get-image :keycard-phone)
+                 :resize-mode :center
+                 :style       {:width  160
+                               :height 170}}]
+   [react/view {:margin-top 10}
+    [react/text {:style {:text-align  :center
+                         :color       colors/gray
+                         :font-size   15
+                         :line-height 22}}
+     (i18n/label :t/hold-card)]]])
+
+(defn- keycard-processing-view []
+  [react/view {:flex-direction  :column
+               :flex            1
+               :justify-content :center
+               :align-items     :center}
+   [react/activity-indicator {:size      :large
+                              :animating true}]
+   [react/text {:style {:margin-top 16
+                        :color      colors/gray}}
+    (i18n/label :t/processing)]])
+
+(defn- sign-with-keycard-button
+  [amount-error gas-error]
+  [button/button {:on-press   #(re-frame/dispatch [:signing.ui/sign-with-keycard-pressed])
+                  :text-style {:padding-right 2
+                               :padding-left  16}
+                  :style      {:background-color colors/black-light
+                               :padding-top      2
+                               :border-radius    8}
+                  :disabled?  (or amount-error gas-error)}
+   (i18n/label :t/sign-with)
+   [react/view {:padding-right 16}
+    [react/image {:source (resources/get-image :keycard-logo)
+                  :style  {:width         64
+                           :margin-bottom 7
+                           :height        26}}]]])
+
+(defn- signing-phrase-view [phrase]
+  [react/view {:align-items :center}
+   [react/text {:style {:color colors/gray :padding-bottom 8}} (i18n/label :t/signing-phrase)]
+   [react/text phrase]])
+
+(defn- keycard-view
+  [{:keys [keycard-step]} phrase]
+  [react/view {:height 450}
```

Comment:
Is the height hardcoded for all possible device sizes?

Reply:
@siphiuel I'm not sure about this one too. But we need this minimum height to render PIN view inside it.

I can put that view into `scroll-view` as possible workaround for devices with height < 450.

https://www.figma.com/file/XUehMnhyD1FGcWzvGz6SXqvh/Mobile-wallet?node-id=1986%3A399&viewport=-1050%2C-2689%2C0.8941199779510498

Reply:
@andmironov @errorists what do you think?